lib/index.js

```diff
@@ -11,13 +12,16 @@ module.exports = (options = {}) => {
 
   const name = 'config-loader';
   const raw = load(options);
+  const schema = merge({}, options.schema, webpackSchema);
 
   return resolve(raw).then((result) => {
-    const schema = merge({}, options.schema, webpackSchema);
-    for (const target of [].concat(result.config)) {
-      validate({ name, schema, target });
-    }
+    const { config, configPath } = result;
 
-    return result;
+    return extend(config).then((finalConfig) => {
+      for (const target of [].concat(finalConfig)) {
```

Comment:
I don't like this trick but since it already was in the source I guess it's fine

Reply:
Are you speaking about the `[].concat`?

Reply:
Yes. It took me some time to understand the intent here and why it was needed. Maybe a comment or an `if` statement before the iteration would have been better.